### PR TITLE
Backport: Gutenberg 4434

### PR DESCRIPTION
**Pull request by scruffian**
This is an attempt at a backport of https://github.com/WordPress/gutenberg/pull/44334

The main part of this change is adding a new CSS file and I'm not sure how to do this - I'm guessing this comes from the packages, rather than me adding it manually.

@ockham any insight would be appreciated.

### CODE REVIEW

**Comment by ockham** [comment]
> This is an attempt at a backport of [WordPress/gutenberg#44334](https://github.com/WordPress/gutenberg/pull/44334)
> 
> The main part of this change is adding a new CSS file and I'm not sure how to do this - I'm guessing this comes from the packages, rather than me adding it manually.

I think that's accurate; [according to unpkg.com](https://unpkg.com/browse/@wordpress/block-library@7.15.0/build-style/), the package currently contains files such as [`common.css`](https://unpkg.com/browse/@wordpress/block-library@7.15.0/build-style/common.css) or [`theme.css`](https://unpkg.com/browse/@wordpress/block-library@7.15.0/build-style/theme.css) at root level, so I'd expect `classic.css` to show up there, too.

I'll be publishing a new round of npms for WP 6.1 Beta 2 later today, and since we've cherry-picked https://github.com/WordPress/gutenberg/pull/44334 to be included with it, we can check unpkg later to see if the file ended up there.

The [path you're using](https://github.com/WordPress/wordpress-develop/pull/3331/files#diff-14577b710f11306118fa14601c82e5831f97b353025bfec3b59d9db9c49ecceaR3651) to then locate the file in Core after a sync and build also matches where I'm seeing those other CSS files locally.

(To recap, we'll need to wait for packages to be published and synced before we can land this PR 😊 )

File: src/wp-includes/script-loader.php [hunk 1]
```diff
@@ -3640,3 +3640,19 @@ function _wp_theme_json_webfonts_handler() {
...
 }
+
+/**
+ * Loads classic theme styles on classic themes.
+ *
+ * This is needed for backwards compatibility for button blocks specifically.
+ */
+function gutenberg_enqueue_classic_theme_styles() {
+	if ( ! wp_is_block_theme() ) {
+		wp_register_style( 'classic-theme-styles', "/wp-includes/css/dist/block-library/classic.css", array(), true );
```
Comment (ockham) [inline src/wp-includes/script-loader.php]: Let's adopt the linter's suggestion here:

```suggestion
		wp_register_style( 'classic-theme-styles', '/wp-includes/css/dist/block-library/classic.css', array(), true );
```

File: src/wp-includes/script-loader.php [hunk 2]
```diff
@@ -3640,3 +3640,19 @@ function _wp_theme_json_webfonts_handler() {
 	add_action( 'wp_enqueue_scripts', $fn_generate_and_enqueue_styles );
 	add_action( 'admin_init', $fn_generate_and_enqueue_editor_styles );
 }
+
+/**
+ * Loads classic theme styles on classic themes.
+ *
+ * This is needed for backwards compatibility for button blocks specifically.
+ */
+function gutenberg_enqueue_classic_theme_styles() {
```
Comment (ockham) [inline src/wp-includes/script-loader.php]: No `gutenberg_` in Core pls 😬 Needs either `wp_` or no prefix at all. (I'm not entirely if there's a rule which of those to choose, but to me, dropping the prefix altogether could be okay here.)

```suggestion
function enqueue_classic_theme_styles() {
```

Comment (ockham) [reply src/wp-includes/script-loader.php]: Hmm, OTOH, we have some precedent for `wp_enqueue_*`. So let's maybe go with that after all.

```suggestion
function wp_enqueue_classic_theme_styles() {
```

File: src/wp-includes/script-loader.php [hunk 3]
```diff
@@ -3640,3 +3640,19 @@ function _wp_theme_json_webfonts_handler() {
...
+	if ( ! wp_is_block_theme() ) {
+		wp_register_style( 'classic-theme-styles', "/wp-includes/css/dist/block-library/classic.css", array(), true );
+		wp_enqueue_style( 'classic-theme-styles' );
+	}
+}
+// To load classic theme styles on the frontend.
+add_action( 'wp_enqueue_scripts', 'gutenberg_enqueue_classic_theme_styles' );
+// To load classic theme styles in the the editor.
+add_action( 'admin_enqueue_scripts', 'gutenberg_enqueue_classic_theme_styles' );
```
Comment (ockham) [inline src/wp-includes/script-loader.php]: See https://github.com/WordPress/wordpress-develop/pull/3331/files#r981195265 and https://github.com/WordPress/wordpress-develop/pull/3331/files#r981196497

```suggestion
add_action( 'wp_enqueue_scripts', 'wp_enqueue_classic_theme_styles' );
// To load classic theme styles in the the editor.
add_action( 'admin_enqueue_scripts', 'wp_enqueue_classic_theme_styles' );
```

**Comment by ockham** [comment]
Published [`@wordpress/block-library@7.14.3`](https://www.npmjs.com/package/@wordpress/block-library/v/7.14.3). It has `classic.css`: https://unpkg.com/browse/@wordpress/block-library@7.14.3/build-style/classic.css.

Here's the sync PR: https://github.com/WordPress/wordpress-develop/pull/3342

File: src/wp-includes/script-loader.php (hunk 1 above)
Comment (ockham) [reply src/wp-includes/script-loader.php]: At second glance, do we need `-$suffix` here (in case we need RTL version)?

**Comment by ockham** [comment]
I'll rebase so this will pick up `classic.css`, now that the [sync PR](https://github.com/WordPress/wordpress-develop/pull/3345) has landed.

File: src/wp-includes/script-loader.php (hunk 1 above)
Comment (scruffian) [reply src/wp-includes/script-loader.php]: Done!

**Review by ockham — APPROVED** [review]
Thank you @scruffian!

Confirming that this enqueues `classic.css` both on the frontend and in the editor ✅ 

LGTM!

File: src/wp-includes/script-loader.php [hunk 4]
```diff
@@ -3640,3 +3640,20 @@ function _wp_theme_json_webfonts_handler() {
...
+	if ( ! wp_is_block_theme() ) {
+		$suffix = wp_scripts_get_suffix();
+		wp_register_style( 'classic-theme-styles', "/wp-includes/css/dist/block-library/classic$suffix.css", array(), true );
+		wp_enqueue_style( 'classic-theme-styles' );
+	}
+}
+// To load classic theme styles on the frontend.
+add_action( 'wp_enqueue_scripts', 'wp_enqueue_classic_theme_styles' );
+// To load classic theme styles in the the editor.
+add_action( 'admin_enqueue_scripts', 'wp_enqueue_classic_theme_styles' );
```
Comment (dream-encode) [inline src/wp-includes/script-loader.php:3659]: I think these should probably be moved to `wp-includes/default-filters.php` where all other core hooks live.  Probably around https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/default-filters.php#L558

Comment (dream-encode) [reply src/wp-includes/script-loader.php:3659]: I'll handle this in the merge though.

Comment (ockham) [reply src/wp-includes/script-loader.php:3659]: Thank you @dream-encode!

**Comment by ockham** [comment]
Some more detail from https://github.com/WordPress/gutenberg/pull/44334's paper trail:

https://github.com/WordPress/gutenberg/pull/43981 has testing instructions:

> ## Testing Instructions
> 1. Switch to a classic theme that doesn't specify its own button styles (I'm using Canape)
> 2. Check that button look broken in trunk
> 3. Check out this PR
> 4. Check that buttons now look correct:
> Screenshot 2022-09-08 at 14 21 57

The styling was apparently originally removed by https://github.com/WordPress/gutenberg/pull/34180, which was [merged](https://github.com/WordPress/gutenberg/commit/b92d56ec43e6701c393ee271c9928c92818419a7) into Gutenberg in June. This means that it likely made its way into Core with the [first package sync for WP 6.1](https://github.com/WordPress/wordpress-develop/commit/3b63a75108b7f6d5e9112c556285ab1bc80ddbcf). Testing this now.

**Comment by ockham** [comment]
> The styling was apparently originally removed by [WordPress/gutenberg#34180](https://github.com/WordPress/gutenberg/pull/34180), which was [merged](https://github.com/WordPress/gutenberg/commit/b92d56ec43e6701c393ee271c9928c92818419a7) into Gutenberg in June. This means that it likely made its way into Core with the [first package sync for WP 6.1](https://github.com/WordPress/wordpress-develop/commit/3b63a75108b7f6d5e9112c556285ab1bc80ddbcf). Testing this now.

Confirming:

| Before 3b63a75108b7f6d5e9112c556285ab1bc80ddbcf | At 3b63a75108b7f6d5e9112c556285ab1bc80ddbcf |
|-------|-------|
| image | image |

**Comment by dream-encode** [comment]
Merged into core in https://core.trac.wordpress.org/changeset/54358.

**Pull request closed without merging**